pkg/types/validation/installconfig.go

```diff
@@ -131,6 +133,15 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
 	if _, ok := validPublishingStrategies[c.Publish]; !ok {
 		allErrs = append(allErrs, field.NotSupported(field.NewPath("publish"), c.Publish, validPublishingStrategyValues))
 	}
+	allErrs = append(allErrs, validateImageDigestSources(c.ImageDigestSources, field.NewPath("imageDigestSources"))...)
+	if _, ok := validPublishingStrategies[c.Publish]; !ok {
+		allErrs = append(allErrs, field.NotSupported(field.NewPath("publish"), c.Publish, validPublishingStrategyValues))
+	}
+	if len(c.ImageContentSources) != 0 && len(c.ImageDigestSources) != 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("imageContentSources"), c.Publish, "cannot set imageContentSources and imageDigestSources at the same time"))
+	} else if len(c.ImageDigestSources) != 0 {
```

Comment:
Shouldn't this check be for  `len(c.ImageContentSources) != 0` not `c.ImageDigestSources` since the warning is indicating that ImageContentSources are being used

Reply:
Also, the else could not be required (ie, always inform the user of the deprecation)